### ISSUE TITLE
Highlight roads not compliant with vehicle parameters

### ISSUE THREAD

**Issue by Max1234-Ita** [osmandapp/OsmAnd#18369]
### Describe the idea (required)

I know that the Routing already excludes the roads that do not comply with vehicle parameters set in the Configure Profile menu; I think it also would be very useful having such roads highlighted directly on the map, in order to provide the driver with a better visual indication.

### Tell us about the expected behaviour (required)

When a road is being rendered on the map, its _width/maxwidth_, _maxweight_, _maxheight_ etc. tags (if any) should be checked against the corresponding vehicle parameters specified for the current profile. 

If the vehicle parameters exceed at least one restriction, then that road should be drawn with a particular style (i.e. highlighted in red), so that the User can immediatley know he/she can't take it, even if Navigation is not active and no route has been calculated.

If the restriction tag is on a single node (this is the case for many _maxwidth_ or _maxheight_ tags), then that node should be highlighted too.

The feature could be turned on or off with a menu option, like it happens, i.e. for highliting the access restrictions in Configure map menu (_ -> Details -> Show access restrictions and toll_ )

### Tell us about alternatives you've considered (required)

Actually none but carefully checking the restrictions on the road signs, (what a driver should always do, actually, indeed).

This feature could be helpful to the users (especially truck or motorhome drivers) by visually providing information with some advance, even without calculating a route.

**Comment by pebogufi** [comment]
I love this idea, great !

**Comment by tradavyvy** [comment]
Maybe also include a maximum steepnes value for streches of road to be marked in the warning highlight color. That would require setting a max. acceptable steepnes for the vehicle (or walking/biking) person to be set with the vehicle parameters.
I like to drive roadtrips without navigation and that would help avoid roads where our motorhome would strugle (or fail).

Make it even perfect with a max. height setting to show passes over a surtain height to avoid winter snow&ice problems.
(This could be used in summer to find roads that are higher up and thus cooler)

**Comment by Max1234-Ita** [comment]
>(...)
> Make it even perfect with a max. elevation setting to show passes/roads over a surtain elevation to avoid winter snow&ice problems. (This could be used in summer to find roads that are higher up and thus cooler)

I don't think this point would be feasible: while _maxwidth_, _maxweight_ etc. tags are quite common on road mapping, the elevation check would require the _ele_ tag to be present on the road node(s), which is a very rare event.

Also steepness check would be quite tough to implement, I think, since it would require to perform an analaysis over groups of road nodes to determine it; also simply checking the presence of the _incline_ tag (https://wiki.openstreetmap.org/wiki/Key:incline) would not help much, since that tag is mostly used with  "up" or "down" values, which doesn't quantify the slope at all (see https://taginfo.openstreetmap.org/keys/incline#values)

**Comment by tradavyvy** [comment]
Incline and elevation are available when navigating in the graph you can 'project' under the map.
But by no means spend excessive amounts of scarce programmers time on this if it is complicated. For it is 'just nice to know information'.
How about giving areas above a certain setable elavation a shading (so not the roads, but the whole map area).

**Comment by Max1234-Ita** [comment]
> Incline and elevation are available when navigating in the graph you can 'project' under the map.
> But by no means spend excessive amounts of scarce programmers time on this if it is complicated. For it is 'just nice to know information'.
> How about giving areas above a certain setable elavation a shading (so not the roads, but the whole map area).

You already can do that!

Just enable the Topography plugin and download the Slope maps for the region of interest 😉
